### PR TITLE
Update ExternalMetrics provider to return error if datadogmetric is stale

### PR DESCRIPTION
**Pull request by sblumenthal**
### What does this PR do?

This PR will cause the external metrics server to return an error if a DatadogMetric object is valid but has not been updated within MaxAge.

### Motivation

See investigation [here](https://app.datadoghq.com/notebook/8582778/dca-serving-stale-datadogmetrics?notebookType=rte)

### Possible Drawbacks / Trade-offs

While this fixes the issue of the WPA scaling based off of metrics that have not been updated recently and have become stale, it does not actually mark the DatadogMetric CR object as invalid, so some confusion could present itself while looking at the cluster.

It also doesn't fix the underlying problem with the current implementation, where slow responses from the API lead to batch failures and batches are broken up and queries are evaluated sequentially, essentially halting all metric evaluation until the API recovers.

### Describe how to test/QA your changes

QA has already been done on my end via a minikube cluster where I spun up ~30 nginx deployments, each one configured to use a WPA for scaling. I modified the DCA using [this change](https://github.com/DataDog/datadog-agent/commit/8066cd80de8e643a396d99fb3d8356204ffd1c80) to simulate slow responses and timeouts, and recorded the results of the scaling behavior before and after the change in [this notebook](https://dddev.datadoghq.com/notebook/8833926/external-metrics-server-metric-staleness-testing)

### CODE REVIEW

**Review by urseberry — APPROVED** [review]

File: pkg/clusteragent/autoscaling/externalmetrics/model/datadogmetricinternal.go [hunk 1]
```diff
@@ -249,11 +259,15 @@ func (d *DatadogMetricInternal) BuildStatus(currentStatus *datadoghq.DatadogMetr
 }
 
 // ToExternalMetricFormat returns the current DatadogMetric in the format used by Kubernetes
-func (d *DatadogMetricInternal) ToExternalMetricFormat(externalMetricName string) (*external_metrics.ExternalMetricValue, error) {
+func (d *DatadogMetricInternal) ToExternalMetricFormat(externalMetricName string, metricsMaxAge int64) (*external_metrics.ExternalMetricValue, error) {
 	if !d.Valid {
 		return nil, fmt.Errorf("DatadogMetric is invalid, err: %v", d.Error)
 	}
 
+	if d.IsStale(metricsMaxAge, time.Now().UTC().Unix()) {
```
Comment (vboulineau) [inline pkg/clusteragent/autoscaling/externalmetrics/model/datadogmetricinternal.go:267]: currentTime should be injected as a parameter instead I believe

**Review by vboulineau — COMMENTED** [review]
We should perhaps increase the default maxAge as we add the time from Controller (HPA,WPA) to ExternalMetrics, which can take several seconds.
Moving to 150s instead of 120s for instance.

**Comment by sblumenthal** [comment]
> We should perhaps increase the default maxAge as we add the time from Controller (HPA,WPA) to ExternalMetrics, which can take several seconds. Moving to 150s instead of 120s for instance.

Synced about this offline, the reasoning behind this makes sense to me, so I updated the default value and added a comment about this change to the changelog

**Review by Kaderinho — APPROVED** [review]

**Review by vboulineau — APPROVED** [review]

**Pull request merged**